### PR TITLE
wally-cli: remove unnecessary line (zsa/wally-cli/pulls/1 has long been m…

### PR DESCRIPTION
**Pull request by johnrichardrinehart**
###### Motivation for this change
An extra line was in an application `default.nix` waiting for a [PR](https://github.com/zsa/wally-cli/pull/1) to be merged. It was merged ~222 days ago.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [N/A ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

*Note* I didn't know how to build this one package alone without using flakes (I ran `nix build '.#legacyPackages.x86_64-linux.wally-cli'`). If someone could tell me how to test the build of this package with `nix-build` (traditional way) I'd really appreciate it.

Thanks!

@spacekookie @r-burns

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122730` at 96da004d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/82e88fc28a9100bc9bb98e59c1f3fa03">wally-cli</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by johnrichardrinehart** [comment]
I've pinged the [wally-cli[(https://github.com/zsa/wally-cli) maintainer to try to fix this build issue. It's just a linting error, though, so we can either:
1) Wait on a new release (trying to contact maintainer)
2) Pin the `rev` in `default.nix` to something later than https://github.com/zsa/wally-cli/commit/0cf3dd904c5d67278d04182debb310c370754da6 .

**Comment by wamserma** [comment]
You can also fetchpatch `https://github.com/zsa/wally-cli/commit/0cf3dd904c5d67278d04182debb310c370754da6.patch`.

**Comment by r-burns** [comment]
I think I like the fetchpatch option best if that's sufficient, since it's the least intrusive. It would be great if the maintainer would create a new release, but we could simply bump to unstable if needed.

**Comment by johnrichardrinehart** [comment]
Let's wait a week to see if we can get a new release. I have a reminder in my calendar to circle back to this.

Good to know about `fetchpatch`, but I don't love it (doesn't look nice).

**Comment by fdidron** [comment]
Hello, 

I'm the maintainer of wally-cli, and would love to help getting this to work.

The lint error is

./dfu.go:180:25: Sprintf call has arguments but no formatting directives

There's no Sprintf directive on line 180 https://github.com/zsa/wally-cli/blob/75e758546d0679f4ce5b562625e3de21b45561f1/dfu.go#L180

There's one on line 181, and it does have a formatting directive.

Regarding the version bump, no worries I'll tag a version once we get that linting issue sorted out.

**Comment by fdidron** [comment]
@johnrichardrinehart Any idea why the CI would point to line 180 where there's not Sprintf directive ?

**Comment by r-burns** [comment]
I think that was fixed by https://github.com/zsa/wally-cli/commit/0cf3dd904c5d67278d04182debb310c370754da6, so once we bump our version or apply that patch we should be good.

**Comment by johnrichardrinehart** [comment]
@fdidron Sorry I didn't explain clearly. This repository (`nixpkgs`) is pinned to the most recent release of `wally-cli`. The most recent release is [227ce](https://github.com/zsa/wally-cli/commit/227ce6f096efc229187d77c97f97d1a73131e6a2) which doesn't yet have the `Sprintf` fix from https://github.com/zsa/wally-cli/commit/0cf3dd904c5d67278d04182debb310c370754da6 . Since this repository requires that the package passes `go vet` the build of `wally-cli` fails.

This package is used in the context of the `nix` package manager. So, this would be equivalent in Debian world to `apt install wally-cli` and it fails because of a linting problem. So, we have the choice, here, to pin to a later commit. But, I thought it might be nice to bump `wally-cli` to 2.0.1, or similar, so we can have a stable release to pin to, here.

I hope that clears things up.

**Comment by fdidron** [comment]
@johnrichardrinehart got it, I tagged a new release https://github.com/zsa/wally-cli/releases/tag/2.0.1-linux

**Comment by johnrichardrinehart** [comment]
@r-burns  and @spacekookie I've added a commit to bump the release to `v2.0.1`. I've tested locally using flakes:

    nix build .#legacyPackages.x86_64-linux.wally-cli

I hope it's cool to include that release bump in this PR.

**Comment by ThibautMarty** [comment]
Does it actually run tests?

```
$ nix-build -A wally-cli
…
github.com/zsa/wally-cli
running tests
?       github.com/zsa/wally-cli        [no test files]
installing
…
```

Yes it's fine to include the version change. Setting `doCheck = true` without bumping to a new release/version (or including a patch) does not make sense.

You can squash the two commits together and please format the new one like [`wally-cli: 2.0.0 -> 2.0.1`](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md)

**Comment by johnrichardrinehart** [comment]
All done.

**Review by r-burns — APPROVED** [review]
Looks good, thanks!

Result of `nixpkgs-review pr 122730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>tests.trivial</li>
    <li>wally-cli</li>
  </ul>
</details>

**Comment by johnrichardrinehart** [comment]
@r-burns @spacekookie @ThibautMarty Are we waiting on anything for the merge and and branch deletion?

**Review by ThibautMarty — APPROVED** [review]

**Pull request merged**
